src/UnderlineNav2/UnderlineNavItem.tsx

```diff
@@ -36,16 +38,21 @@ export type UnderlineNavLinkProps = {
    */
   leadingIcon?: React.FunctionComponent<IconProps>
   as?: React.ElementType
+  /**
+   * Counter
+   */
+  counter?: number
```

Comment:
@danielguillan - Wondering if we are going to support loading state for counters.

Reply:
Just making a comment here for the types. I added the counter prop with only number type just to make the Counter component redundant. We can add another type for the loading state if we are going to support 😊

Reply:
Yes, we are going to support loading states. We can add support incrementally in another PR if that makes things easier.